tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -256,7 +256,14 @@ public:
    // TODO implement MakeNew. Requires some smartness in passing the appropriate previous node.
 };
 
-class FillHelper : public RActionImpl<FillHelper> {
+/// This helper fills TH1Ds for which no axes were specified by buffering the fill values to pick good axes limits.
+///
+/// TH1Ds have an automatic mechanism to pick good limits based on the first N entries they were filled with, but
+/// that does not work in multi-thread event loops as it might yield histograms with incompatible binning in each
+/// thread, making it impossible to merge the per-thread results.
+/// Instead, this helper delays the decision on the axes limits until all threads have done processing, synchronizing
+/// the decision on the limits as part of the merge operation.
+class DelayedFillHelper : public RActionImpl<DelayedFillHelper> {
```

Comment:
It's delayed - but the crucial property isn't the delayed filling, it's *buffered*, and it's buffered to determine the axis limits. Possible other names could thus be
- `BufferedFillHelper`
- `AutoAxisRangeFillHelper`
- `AutoRangeFillHelper`

Reply:
good point, will change to `BufferedFillHelper`